pennylane/ops/functions/evolve.py

```diff
@@ -14,32 +14,64 @@
 """
 This module contains the qml.evolve function.
 """
-from typing import Union
+import warnings
+from functools import singledispatch
 
 from pennylane.operation import Operator
 from pennylane.ops import Evolution
 from pennylane.pulse import ParametrizedEvolution, ParametrizedHamiltonian
 
 
-def evolve(op: Union[Operator, ParametrizedHamiltonian]):
-    """Returns a new operator that computes the evolution of ``op``.
+@singledispatch
+def evolve(*args, **kwargs):  # pylint: disable=unused-argument
+    r"""This method is dispatched and its functionality depends on the type of the input ``op``.
+
+    .. raw:: html
+
+        <html>
+            <h3>Input: Operator</h3>
+            <hr>
+        </html>
+
+    Returns a new operator that computes the evolution of ``op``.
+
+    .. math::
+
+        e^{-i \times \bm{op} \times coeff)}
```

Comment:
```suggestion
        e^{-i \cdot \bm{\text{op}} \cdot \text{coeff}}
```

Reply:
Or maybe \text{exp}(-i \cdot \bm{\text{op}} \cdot \text{coeff}) ? The text gets so small with e^ 

But for sure the close-parenthesis after coeff should go, the rest is pretty subjective.

Reply:
I decided to not use any symbol for the multiplication: `e^{-i x \bm{O})}`

I changed `coeff` and `op` by `x` and `O` to make it more visual. 

What do you think?

Reply:
Nvm looks like you already changed it :D

Reply:
I think that seems good :)